react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx

```diff
@@ -1,40 +1,87 @@
 /*!
- * Copyright 2023 Cognite AS
+ * Copyright 2024 Cognite AS
  */
-import { useEffect, type ReactElement } from 'react';
+import { useEffect, type ReactElement, useState } from 'react';
 
-import { CogniteCadModel, type CogniteModel } from '@cognite/reveal';
+import {
+  CogniteCadModel,
+  TreeIndexNodeCollection,
+  type CogniteModel,
+  NumericRange,
+  type NodeAppearance
+} from '@cognite/reveal';
 import { useAllMappedEquipmentAssetMappings } from '../..';
 import { Color } from 'three';
-import { type RuleOutputSet } from './types';
+import { type AssetStylingGroupAndStyleIndex } from './types';
 import { generateRuleBasedOutputs } from './utils';
-import { type FdmPropertyType } from '../Reveal3DResources/types';
 import { use3dModels } from '../../hooks/use3dModels';
+import { type AssetMapping3D } from '@cognite/sdk/dist/src';
+import { filterUndefined } from '../../utilities/filterUndefined';
+import { type AssetStylingGroup, type FdmPropertyType } from '../Reveal3DResources/types';
 
 export type ColorOverlayProps = {
-  ruleSet: RuleOutputSet | Record<string, any> | FdmPropertyType<Record<string, any>> | undefined;
+  ruleSet: Record<string, any> | FdmPropertyType<Record<string, any>> | undefined;
+  onRuleSetChanged?: (currentStylings: AssetStylingGroupAndStyleIndex[] | undefined) => void;
 };
 
-export function RuleBasedOutputsSelector({ ruleSet }: ColorOverlayProps): ReactElement | undefined {
+export function RuleBasedOutputsSelector({
+  ruleSet,
+  onRuleSetChanged
+}: ColorOverlayProps): ReactElement | undefined {
   const models = use3dModels();
 
+  const [stylingGroups, setStylingsGroups] = useState<AssetStylingGroupAndStyleIndex[]>();
+
   const {
     data: assetMappings,
     isFetching,
     hasNextPage,
     fetchNextPage
   } = useAllMappedEquipmentAssetMappings(models);
 
+  const applyBasedNodeStyling = (
+    model: CogniteModel,
+    assetMappings: AssetMapping3D[]
+  ): AssetStylingGroupAndStyleIndex | undefined => {
+    if (!(model instanceof CogniteCadModel)) {
+      return;
+    }
+
+    const baseNodeStyling = new TreeIndexNodeCollection();
+    assetMappings?.forEach((assetMapping) => {
+      const range = new NumericRange(assetMapping.treeIndex, assetMapping.subtreeSize);
+      baseNodeStyling.getIndexSet().addRange(range);
+    });
+    const nodeAppearance: NodeAppearance = {
+      color: new Color('#efefef')
+    };
+    const assetStylingGroup: AssetStylingGroup = {
+      assetIds: assetMappings.map((node) => node.assetId),
+      style: { cad: nodeAppearance }
+    };
+
+    const baseStylingGroup: AssetStylingGroupAndStyleIndex = {
+      styleIndex: baseNodeStyling,
+      assetStylingGroup
+    };
+
+    return baseStylingGroup;
+  };
+
```

Comment:
I think a better way to do this, would be to disable the `mapped`-styling in Search. We specify a `defaultResourceStyling`-property in the `SceneContainer` in `RevealContent.tsx` in Fusion. We can override this to disable the mapped-color there instead. Sorry, I realize I should have mentioned this earlier :sweat:

Reply:
Ah yeah. I forgot to have a look on that. Changed here:
https://github.com/cognitedata/reveal/pull/4296/commits/553ce2a145f6f71b97d84ac9ee189bfed00381dc